File: 240124/되돌아오기/come-back.py
n = int(input())
x, y = 0, 0
dx = [1, 0, -1, 0]
dy = [0, -1, 0, 1]
d = {
    'E': 0,
    'S': 1,
    'W': 2,
    'N': 3
}
cnt = 0
answer = -1

for i in range(n):
    dir, dist = input().split()
    dir_num, dist = d[dir], int(dist)

    for _ in range(dist):
        x += dx[dir_num]
        y += dy[dir_num]
        cnt += 1

        if x == 0 and y == 0:
            answer = cnt
            break
            
    if x == 0 and y == 0:
        break

print(answer)

File: 240912/되돌아오기/come-back.js
const fs = require("fs");
let input = fs.readFileSync(0).toString().trim().split("\n");

let n = Number(input[0]);
let mapper = {
    "W": 0, // 서
    "S": 1, // 남
    "N": 2, // 북
    "E": 3  // 동
};

let dx = [-1, 0, 0, 1], dy = [0, -1, 1, 0];
let [x, y] = [0, 0];
let cDir = mapper[0];
let time = 0;
let satisfied = false;

outer: for (let i = 1; i < input.length; i++) {
    let [dir, dist] = input[i].trim().split(" ");
    dist = Number(dist);
    cDir = mapper[dir];
    for (let j = 0; j < dist; j++) {
        x += dx[cDir]; y += dy[cDir];
        time++;
        
        if (x === 0 && y === 0) {
            satisfied = true;
            break outer;
        }
    }
}

if (satisfied) {
    console.log(time);
} else {
    console.log(-1);
}